src/wasm/wasm-stack.cpp

```diff
@@ -1910,7 +1910,19 @@ void BinaryInstWriter::visitStructNew(StructNew* curr) {
 }
 
 void BinaryInstWriter::visitStructGet(StructGet* curr) {
-  WASM_UNREACHABLE("TODO (gc): struct.get");
+  const auto& heapType = curr->value->type.getHeapType();
+  const auto& field = heapType.getStruct().fields[curr->index];
+  int8_t op;
+  if (field.type != Type::i32 || field.packedType == Field::not_packed) {
+    op = BinaryConsts::StructGet;
+  } else if (curr->signed_) {
+    op = BinaryConsts::StructGetS;
+  } else {
+    op = BinaryConsts::StructGetU;
+  }
+  o << int8_t(BinaryConsts::GCPrefix) << int8_t(op);
```

Comment:
The `op` should be a uleb128 here. (Not that it makes a difference in practice, but it would be good to be consistent with the spec for readability purposes.)

Reply:
It looks like all the other locations use `int8_t`, though. But I agree this might be a good cleanup for the whole file - later?

Reply:
Specifically only _prefixed_ opcodes should be LEBs, but I see that you're right that we are inconsistent here. All the SIMD opcodes use LEBs but e.g. none of the atomics instructions use LEBs. Cleaning them all up in a separate PR sgtm.